cluster/src/cluster_impl.rs

```diff
@@ -272,7 +273,43 @@ impl Inner {
             msg: "current shard info is missing in UpdateShardInfo",
         })?;
         let table_info = req.table_info.clone().context(TableNotFound {
-            msg: "table info is missing in CreateTableOnShardRequest",
+            msg: "table info is missing in DropTableOnShardRequest",
+        })?;
+
+        self.shard_tables_cache.try_remove_table_from_shard(
+            update_shard_info.prev_version,
+            ShardInfo::from(curr_shard_info),
+            TableInfo::from(table_info),
+        )
+    }
+
+    fn open_table_on_shard(&self, req: &OpenTableOnShardRequest) -> Result<()> {
```

Comment:
The implementation of `open_table_on_shard` and `close_table_on_shard` seems to be exactly the same as `create_table_on_shard` and `drop_table_on_shard`. Maybe it's better to encapsulate their implementation as a public method.